codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/model/knowledge/EndpointParameterIndex.kt

```diff
@@ -44,14 +46,23 @@ class EndpointParameterIndex private constructor(model: Model) : KnowledgeIndex
             }
         }
 
+    /**
+     * Get the [operationContextParams](https://smithy.io/2.0/additional-specs/rules-engine/parameters.html#smithy-rules-operationcontextparams-trait)
+     * for an operation.
+     *
+     * @param op the operation shape to get context params for.
+     */
+    fun operationContextParams(op: OperationShape): MutableMap<String, OperationContextParamDefinition>? =
+        op.getTrait<OperationContextParamsTrait>()?.parameters
```

Comment:
**Question**: Why does this need to return a _mutable_ list?

Reply:
This shouldn't be mutable, this was my mistake